### DISCUSSION TITLE
Animations optimised for CoreAnimation rendering

### DISCUSSION THREAD

**Discussion by Hydralo** [airbnb/lottie-ios#2083] [Q&A]
I recently encountered some significant performance issues after migrating from Lottie version 2.5.3 to version 4.2.0. These issues seem to be related to rendering changes in rendering engine.

To overcome this hurdle, I kindly request the assistance of the developer community in providing a pack of animations that have been thoroughly optimized for rendering using Lottie version 4.2.0 and the CoreAnimation engine. The goal is to test animations performance in ideal conditions inserting them directly into project assets and then using them in collection view cells.

In particular, I am interested in animations that have been tested and confirmed to perform well with Lottie 4.2.0 and CoreAnimation engine. Any examples, best practices, or code snippets would be immensely helpful in improving the performance of my app and delivering an exceptional user experience.

**Comment by mkj-is** [comment]
The main reason for a slow Lottie animation initialization is the number of layers and animations with a large amount of keyframes. There is a nice discussion going on over at https://github.com/airbnb/lottie-ios/discussions/1886#discussioncomment-8666157.
